src/goto-harness/recursive_initialization.cpp

```diff
@@ -763,3 +817,52 @@ code_blockt recursive_initializationt::build_dynamic_array_constructor(
 
   return body;
 }
+
+bool recursive_initializationt::needs_freeing(const exprt &expr) const
+{
+  if(expr.type().id() != ID_pointer || expr.type().id() == ID_code)
```

Comment:
This could just be `return expr.type().id() = ID_pointer && expr.type().id() != ID_code;`

Reply:
Done.